api/src/pcapi/alembic/versions/20250124T094645_65d6cbee26d7_unicity_chronicle_formid.py

```diff
@@ -0,0 +1,57 @@
+"""enforce formI9d unicity for chronicles"""
+
+from alembic import op
+
+from pcapi import settings
+
+
+# pre/post deployment: post
+# revision identifiers, used by Alembic.
+revision = "65d6cbee26d7"
+down_revision = "5450a88dc12b"
+branch_labels: tuple[str] | None = None
+depends_on: list[str] | None = None
+
+
+def upgrade() -> None:
+    op.execute(
+        """
+DELETE FROM
+    chronicle as c
+WHERE
+    -- delete when more than one is present
+    c."formId" IN (
+        SELECT 
+            form_id
+        FROM (
+            SELECT 
+                "formId" AS form_id, 
+                count(id) AS count_id 
+            FROM 
+                chronicle 
+            GROUP BY 
+                "formId"
+        ) AS count_sub WHERE count_id>1 
+    )
+    -- keep one from delete
+    AND c.id != (
+        SELECT
+            id
+        FROM 
+            chronicle
+        WHERE
+            "formId"=c."formId"
```

Comment:
Question/suggestion : serait-il pertinent, logique, de garder explicitement le premier créé ?
```suggestion
            "formId"=c."formId"
        ORDER BY id
```

Reply:
en vrai on s'en fiche un peu, ils sont tous identiques, mais ça coûte pas grand chose de le faire